sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs/src/Config/BlobsExtensionConfigProvider.cs

```diff
@@ -337,7 +361,44 @@ private BlobClient ConvertBlobBaseClientToBlobClient(BlobBaseClient input, BlobT
             }
         }
 
-        private async Task<Stream> CreateStreamAsync(
+        private Task<Stream> CreateStreamAsync(
+            BlobAttribute blobAttribute,
+            ValueBindingContext context)
+        {
+            if (_functionDataCache != null)
```

Comment:
should we check some runtime value instead of this to detect where this code is running?
I'd rather not use absence of that cache to tell where we are. Is IFunctionDataCache a public type? - if yes then user of the extension can trigger this logic by providing own imlementation.